src/native/external/README.md

```diff
@@ -0,0 +1,72 @@
+# Native external libraries
+
+This folder contains the source code of vendored third party native libraries that are used as dependencies for dotnet/runtime.
+
+### Folder structure
+
+Each native library is roughly structured the following way:
+
+```
+library-folder/
+    library-source-code-files
+library.cmake
+library-version.txt
+cgmanifest.json
+```
+
+- `library-folder/` is where the native source code lives.
+- `library.cmake` is the main cmake file we add to build this library from anywhere in this repo.
+- `library-version.txt` contains all the detailed release information for this library.
+- `cgmanifest.json` is the official file that describes the source code provenance for each one of the external libraries we consume.
+
+### How to add or update a vendored library
+
+1. Download a copy of the source code from an official public release and extract it inside the library folder under `src/native/external/<library-name>`.
+
+   - If the source code comes from a public github repo, you would download it from an url like: `https://github.com/org/repo/releases`.
+   - Make a note of the commit used to snap that release.
+   - From the root folder of the repo, make a note of the license file.
+
+2. Open the `cgmanifest.json` file. Add or modify the entry for this library to indicate the commit hash from which the copy of this release was obtained. The entry looks like this:
+
+```json
+"Registrations": [
+  {
+    "Component": {
+      "Type": "git",
+      "Git": {
+        "RepositoryUrl": "https://github.com/org/repo",
+        "CommitHash": "<commit from which the source code was obtained>"
+      }
+    }
+  }
+]
+```
+
+3. Open the [THIRD-PARTY-NOTICES.txt](https://github.com/dotnet/runtime/blob/main/THIRD-PARTY-NOTICES.TXT) file in the root folder of the runtime repo. Add or update the copy of the license of this library, making sure the license copy has a header like this:
+
+```
+License notice for <library name>
+-----------------------
+```
+
+4. Add or edit the `<library>-version.txt` file under `src/native/external`. This file should contain:
+  - The release version of the downloaded release
+  - The hash commit of the downloaded release
+  - The URL to the tag of the downloaded release in the format `https://github.com/org/repo/releases/tag/<version_number>`
+  - An optional section describing any additional information like:
+    - Manual modifications we need to make after copying the source code. For example, deleting unnecessary files or trimming unnecessary code.
+    - Important notes about the release, like security fixes.
+
+5. Make any necessary cmake changes to properly consume the source code.
```

Comment:
It is not unusual to make changes in the library sources as part of this step. The information about these changes should be included in ...-version.txt file. The same changes should be proposed for the library upstream so that they are not necessary during the next library update.